csrc/core/determinism.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+#include <cstdlib>
+#include <string>
+#include <cctype>
+
+namespace vllm {
+
+// Utility: convert "foo_bar" -> "VLLM_KERNEL_OVERRIDE_DETERMINISM_FOO_BAR"
+inline std::string make_det_env_name(const char* key) {
+  std::string env = "VLLM_KERNEL_OVERRIDE_DETERMINISM_";
+  for (const char* p = key; *p; ++p) {
+    env.push_back(std::toupper(static_cast<unsigned char>(*p)));
+  }
+  return env;
+}
```

Comment:
I don't think we need to be fine-grained on the control here. We just need a global mode -- batch invariant and batch variant. Because as long as one op is batch variant, even if the other op is batch invariant the final result will be batch variant.

Reply:
done